modules/common/logging/hw-mac-retrieve.nix

```diff
@@ -0,0 +1,31 @@
+# Copyright 2022-2024 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}: let
+  cfg = config.ghaf.logging.client;
+  # TODO: replace sshCommand and MacCommand with givc rpc to retrieve Mac Address
+  sshCommand = "${pkgs.sshpass}/bin/sshpass -p ghaf ${pkgs.openssh}/bin/ssh -o StrictHostKeyChecking=no ghaf@net-vm";
+  MacCommand = "cat /sys/class/net/wlp0s5f0/address";
+in {
+  config = lib.mkIf cfg.enable {
+    # TODO: Remove hw-mac.service and replace with givc rpc later
+    systemd.services."hw-mac" = {
+      description = "Retrieve MAC address from net-vm";
+      wantedBy = ["alloy.service"];
+      requires = ["network-online.target"];
+      serviceConfig = {
+        Type = "oneshot";
+        RemainAfterExit = "yes";
+        # Make sure we can ssh before we retrieve mac address
```

Comment:
Ah, the unconditional redirection to the output file is the problem.

We could use pkgs.writers.writeBash, buffer the output and only write on success, but that's probably ok to do in a followup too.

Reply:
Sure, This `hw-mac.service` is temporary code, I have added TODO as well, It will be soon replaced with `givc` implementation. I prefer not to spend much efforts here.